tests/PrettyParseTests.hs

```diff
@@ -108,21 +108,25 @@ genExpr =
     \(Size n) -> Fix <$>
       if n < 2
         then Gen.choice [genConstant, genStr, genSym, genLiteralPath, genEnvPath]
-        else Gen.frequency
+        else
+          let
+            sizeBy i = Size $ n `div` i
+          in
+          Gen.frequency
           [ (1 , genConstant)
           , (1 , genSym)
-          , (4 , Gen.resize (Size (n `div` 3)) genIf)
+          , (4 , Gen.resize (sizeBy 3) genIf)
           , (10, genRecSet)
           , (20, genSet)
           , (5 , genList)
           , (2 , genUnary)
-          , (2 , Gen.resize (Size (n `div` 3)) genBinary)
-          , (3 , Gen.resize (Size (n `div` 3)) genSelect)
-          , (20, Gen.resize (Size (n `div` 2)) genAbs)
-          , (2 , Gen.resize (Size (n `div` 2)) genHasAttr)
-          , (10, Gen.resize (Size (n `div` 2)) genLet)
-          , (10, Gen.resize (Size (n `div` 2)) genWith)
-          , (1 , Gen.resize (Size (n `div` 2)) genAssert)
+          , (2 , Gen.resize (sizeBy 3) genBinary)
+          , (3 , Gen.resize (sizeBy 3) genSelect)
+          , (20, Gen.resize (sizeBy 2) genAbs)
+          , (2 , Gen.resize (sizeBy 2) genHasAttr)
+          , (10, Gen.resize (sizeBy 2) genLet)
+          , (10, Gen.resize (sizeBy 2) genWith)
+          , (1 , Gen.resize (sizeBy 2) genAssert)
```

Comment:
Could go even further with `downscaleBy = Gen.resize . Size . div n` and `(1 , downscaleBy 2 genAssert)`, or even map downscaleBy on curried nested tuples like `(1 , (2, genAssert))`. Am I going too far ? :thinking: :smile:

Reply:
Oh, already outdated

Reply:
It is not:

![Screenshot-2021-10-14-13:51:46](https://user-images.githubusercontent.com/20933385/137311048-dde416b0-d838-495c-a2f3-4e061b52789f.png)

The "outdated" is a purely technical label. (I think) GH generally doesn't hide the stuff until it seems to get addressed.

You are right.

I try to practice the PR review scoping more, generally, the best way to do the breather PR review for both sides is to address only the scope under the PR diffs, not above or around it. Additionally, everything probably should go with the word "please" to a contributor.

Like here - I did refactor, but you say "the more refactor can be done", which is true - but not the scope of the code given, I didn't write the code/form that was asked to refactor.

The refactoring is most effective to enforce on the initial code PR author. But ... it is a peculiar situation to tell a person "rewrite your own code in a better way", especially when they volunteer. So first merging the code, then somebody else refactoring it - seems like a lot easier natural strategy.

I generally like to do this stuff incrementally in the next PRs/commits.